tls/s2n_signature_algorithms.c

```diff
@@ -173,11 +202,13 @@ int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn,
     if (peer_wire_prefs != NULL && peer_wire_prefs->len > 0) {
         int result = s2n_choose_sig_scheme(conn, peer_wire_prefs, &chosen_scheme);
 
-        /* We require an exact match in TLS 1.3, but all previous versions can fall back to the default.
-         * The pre-TLS1.3 behavior is an intentional choice to maximize support. */
-        S2N_ERROR_IF(result != S2N_SUCCESS && conn->actual_protocol_version == S2N_TLS13,
-                S2N_ERR_INVALID_SIGNATURE_SCHEME);
+        /* If we can't find a match in TLS 1.3, we pick the best signature algorithm to use */
+        /* https://tools.ietf.org/html/rfc8446#section-4.4.2.2 */
+        if (conn->actual_protocol_version == S2N_TLS13 && result != S2N_SUCCESS) {
+            GUARD(s2n_tls13_preferred_sig_scheme(conn, &chosen_scheme));
+        }
```

Comment:
Rather than changing this logic, maybe update "s2n_choose_default_sig_scheme" to support TLS1.3.

Reply:
I initially thought about doing that but wasn't exactly happy that s2n_choose_default_sig_scheme() would have to do more conditionals in there although I'm not opposed to doing that.